mso/resource_mso_schema_template_contract_filter.go

```diff
@@ -72,753 +75,253 @@ func resourceMSOTemplateContractFilter() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
 			},
-
 			"directives": {
-				Type:     schema.TypeList,
-				Elem:     &schema.Schema{Type: schema.TypeString},
-				Required: true,
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"none",
+						"no_stats",
+						"log",
+					}, false),
+				},
+				Optional: true,
+				Computed: true,
```

Comment:
How does this work with existing state do we need a state migration?

Reply:
I have tested by creating a configuration and using the current provider to push the configuration. Then execute upgrade of the provider without any changes. This worked. Then made changes to the set, which were applied correctly. Then changed order also to check no changes occur because of set type.